tests/Pitch/EnharmonicTests.elm

```diff
@@ -0,0 +1,59 @@
+module EnharmonicTests exposing (all)
+
+import Expect
+import Fuzz exposing (Fuzzer)
+import MusicTheory.Internal.Pitch as Internal exposing (PitchError(..))
+import MusicTheory.Internal.PitchClass as PitchClass
+import MusicTheory.Letter exposing (Letter(..))
+import MusicTheory.Octave as Octave exposing (OctaveError(..))
+import MusicTheory.Pitch as Pitch
+import MusicTheory.Pitch.Enharmonic as PitchEnharmonic
+import MusicTheory.PitchClass.Enharmonic as PitchClassEnharmonic
+import Test exposing (..)
+import Util.OctaveFuzzer
+import Util.PitchClassFuzzer
+
+
+all : Test
+all =
+    describe "Pitch Tests"
+        [ fuzz2 Util.PitchClassFuzzer.pitchClass Util.OctaveFuzzer.octave "simple enharmonic should have same number of semitones" <|
+            \pitchClass octave ->
+                let
+                    pitch =
+                        pitchClass |> Internal.fromPitchClass octave
+
+                    expectedSemitones =
+                        pitch |> Internal.semitones
+
+                    expectedResult =
+                        if expectedSemitones < 0 then
+                            Err <| InvalidEnharmonicEquivalent (pitch |> Internal.pitchClass |> PitchClassEnharmonic.simple) (BelowValidRange -1)
+
+                        else if expectedSemitones >= 108 then
+                            Err <| InvalidEnharmonicEquivalent (pitch |> Internal.pitchClass |> PitchClassEnharmonic.simple) (AboveValidRange 9)
+
+                        else
+                            Ok expectedSemitones
+                in
+                pitch
+                    |> PitchEnharmonic.simple
+                    |> Result.map Internal.semitones
+                    |> Expect.equal expectedResult
+        , test "enharmonic of Ebbb4 as natural or sharp should be C#4" <|
```

Comment:
"Enharmonic" is an adjective, so it doesn't really make sense to say "an enharmonic of Ebbb4". We should standardize on "enharmonic spelling" or "enharmonic equivalent"